src/components/Chart/Chart.tsx

```diff
@@ -163,17 +168,60 @@ class Chart extends React.Component<
       this.setState({
         chartReady: true
       });
+
+      this.addPriceTypeButton();
     });
   };
 
+  addPriceTypeButton = () => {
+    const {selectedPriceType, selectPriceType} = this.props;
+
+    const buttonProps = {
+      [PriceType.Mid]: {
+        cssClass: 'button_trade',
+        title: 'Trades',
+        onClick: () => {
+          selectPriceType!(PriceType.Trade);
+        }
+      },
+      [PriceType.Trade]: {
+        cssClass: 'button_mid',
+        title: 'Mid price',
+        onClick: () => {
+          selectPriceType!(PriceType.Mid);
+        }
+      }
+    };
+
+    if (selectedPriceType) {
+      const button = buttonProps[selectedPriceType!];
+
+      this.tvWidget
+        .createButton()
+        .attr('title', button.title)
+        .attr('class', `button ${button.cssClass}`)
+        .on('click', button.onClick)
+        .append(' ');
+    }
+  };
+
   saveSettings = () => {
-    this.tvWidget.save(this.props.saveSettings);
+    this.tvWidget.save((settings: any) => {
+      settings.priceType = this.props.selectedPriceType;
+      this.props.saveSettings!(settings);
+    });
   };
 
   loadSettings = async () => {
     await this.props.loadSettings!()
       .then((res: any) => {
-        this.tvWidget.load(JSON.parse(res.Data));
+        const settings = JSON.parse(res.Data);
+
+        this.tvWidget.load(settings);
```

Comment:
not sure the chart will load its settings automatically on the page load, so the _price type_ will not be applied on the initial load. may be wrong